packages/block-editor/src/components/create-new-post-link/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+const CreateNewPostLink = () => {
+	const postType = useSelect( ( select ) => {
+		const { getSelectedBlock } = select( blockEditorStore );
+		return getSelectedBlock()?.attributes?.query?.postType;
```

Comment:
Maybe we can pass the clientID in as a prop, rather than using the selected block.

If it were ever possible to show the inspector of a non-selected block, at which point this would be buggy.

Reply:
Thanks, I've made the suggested change in https://github.com/WordPress/gutenberg/pull/49566/commits/b62b2c1a7bf5e5b26d1a64478184a9ac73091be1.